docs/feature-requests/005-blank-slate.md

```diff
@@ -0,0 +1,265 @@
+* Adds the clone destination path as a project root.
+* Ensures that the clone destination path is the active GitHub package context.
+* Closes the "Create repository" dialog.
+
+### "Local repository" mode
+
+The major difference in this mode is that certain controls are pre-populated with values from the state of the provided local repository.
+
+* The "repository name" field is pre-populated with the directory name of the local repository's root directory.
+* The "source remote" field is invalid if a remote with the given name is already present in the local repository.
+
+Clicking the "Create" button also behaves slightly differently:
+
+* Initializes a git repository in the local repository path if it is not already a git repository.
+* Creates a repository on GitHub with the chosen owner, name, and description.
+* Adds a remote with the specified "source remote name" and sets it to the clone URL of the newly created repository, respecting the https/ssh toggle.
+* If a branch called `master` is present in the local repository, its push and fetch upstreams are configured to be the source remote.
+* The local repository path is added as a project root if it is not already present.
+* Ensures that the clone destination path is the active GitHub package context.
+* Closes the "Create repository" dialog.
+
+## Improved branch publish behavior
+
+If a remote is present in the current repository with a name matching the setting `github.cloneSourceRemoteName`, both clicking "publish" in the push-pull status bar tile and clicking a "publish ..." button in the GitHub tab push HEAD to the clone source remote instead of `origin`, even if the "chosen" remote differs.
+
+If a multiple remotes are present in the current repository, and one is present with a name matching the setting `github.upstreamRemoteName` that has a recognized GitHub URL, it will be preferred as the default remote by the `GitTabContainer` component. Otherwise, if one is present with a name matching the setting `github.cloneSourceRemoteName` and a GitHub URL, that one will be used. Finally we'll fall back to our existing `RemoveSelectorView` menu.
+
+When multiple remotes are present in the current repository and the push-pull status bar tile is in its "publish" state, the push-pull status bar tile's context menu includes a separate "Push" entry for each available remote.
+
+**_Part 2 - Additional information_**
+
+## :anchor: Drawbacks
+
+Modal dialogs are disruptive to UX flow. You can't start creating a repository, have another thought and make a quick edit, then come back to it. This design uses a lot of them.
+
+The "Create repository" flow is missing some of the functionality that the dotcom page has, like initializing a README and a license. We can make _some_ things nicer with the local context we have to work with - like guessing a repository name from the project directory - but we'd be unlikely to keep up with what's available on dotcom.
```

Comment:
Interesting!

I put this here because I actually started to write them in, and I thought it might be stretching too thin, hehe. They aren't _trivial_ - we need to handle fetching valid licenses from the choosealicense API, templating in the current year and stuff, and reading the available gitignores from the github/gitignore repo - but also not super bad. And personally I'm always a bit annoyed about not being able to use them when you start a repo locally instead of on dotcom.

I'll add 'em back in and see what we think 👍

Reply:
Alas, the [`createRepository` GraphQL mutation](https://developer.github.com/v4/input_object/createrepositoryinput/) doesn't yet support the README, gitignore, or license options. :sad: I'll continue to punt on these for this iteration.

I still really like the idea of integrating them later. Maybe even as general options when you already have a repository.